File: src/test/java/lab1/VectorTest.java
package lab1;

import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.*;


class VectorTest {

    private Vector vectorA;
    private Vector vectorB;

    @Test
    void length() {
        vectorA = new Vector(1, 2, 3);
        assertEquals(3.7416573867739413, vectorA.length());

        vectorB = new Vector(-2, 7, 9);
        assertEquals(11.575836902790225, vectorB.length());

    }

    @Test
    void scalarProduct() {
        vectorA = new Vector(12, 36, 77);
        vectorB = new Vector(10, 7, 13);

        assertEquals(1373, vectorA.scalarProduct(vectorB));
    }

    @Test
    void vectorProduct() {
        vectorA = new Vector(89, 69, 7);
        vectorB = new Vector(1, 2, 3);

        Vector result = vectorA.vectorProduct(vectorB);
        assertEquals(193, result.getX());
        assertEquals(-260, result.getY());
        assertEquals(109, result.getZ());
    }

    @Test
    void cos() throws Exception {
        vectorA = new Vector(0, 0, 0);
        vectorB = new Vector(5, 5, 5);

        assertThrows(ArithmeticException.class, () -> vectorA.cos(vectorB));

        Vector vectorC = new Vector(2, 4, 5);
        Vector vectorD = new Vector(4, 7, 10);

        assertEquals(0.9980452274793343, vectorC.cos(vectorD));
    }

    @Test
    void cosToRadian() throws Exception {
        vectorA = new Vector(2, 4, 5);
        vectorB = new Vector(4, 7, 10);

        double degree = vectorA.cosToRadian(vectorA.cos(vectorB));
        assertEquals(3.5830800741464497, degree);
    }

    @Test
    void add() {
        vectorA = new Vector(19, 18, 17);
        vectorB = new Vector(-3, -2, -1);

        Vector result = vectorA.add(vectorB);
        assertEquals(16, result.getX());
        assertEquals(16, result.getY());
        assertEquals(16, result.getZ());
    }

    @Test
    void sub() {
        vectorA = new Vector(-1, 3, 6);
        vectorB = new Vector(-1, -2, -9);

        Vector result = vectorA.sub(vectorB);
        assertEquals(0, result.getX());
        assertEquals(5, result.getY());
        assertEquals(15, result.getZ());
    }

    @Test
    void generateRandomVectors() {
        int countVectors = 5;
        vectorA = new Vector();
        Vector[] vectors = vectorA.generateRandomVectors(countVectors);

        assertEquals(countVectors, vectors.length);
    }
}

File: src/test/java/VectorTest.kt
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import kotlin.ArithmeticException

internal class VectorTest {

    @Test
    fun length() {
        val expectedLength = 3.7416573867739413
        assertEquals(expectedLength, Vector(1.0, 2.0, 3.0).length())
    }

    @Test
    fun scalarProduct() {
        val expectedScalarProduct = 1373.0
        val a = Vector(12.0, 36.0, 77.0)
        val b = Vector(10.0, 7.0, 13.0)
        assertEquals(expectedScalarProduct, a.scalarProduct(b))
    }

    @Test
    fun vectorProduct() {
        val expectedVector = Vector(193.0, -260.0, 109.0)
        val a = Vector(89.0, 69.0, 7.0)
        val b = Vector(1.0, 2.0, 3.0)
        val result = a.vectorProduct(b)
        assertEquals(expectedVector.x, result.x)
        assertEquals(expectedVector.y, result.y)
        assertEquals(expectedVector.z, result.z)
    }

    @Test()
    fun cos() {
        val a = Vector(0.0, 0.0, 0.0)
        val b = Vector(5.0, 5.0, 5.0)
        assertThrows(ArithmeticException::class.java) {
            a.cos(b)
        }

        val expectedCos = 0.9980452274793343
        val c = Vector(2.0, 4.0, 5.0)
        val d = Vector(4.0, 7.0, 10.0)
        assertEquals(expectedCos, c.cos(d))
    }

    @Test
    fun cosToRadian() {
        val expectedDegree = 3.5830800741464497
        val a = Vector(2.0, 4.0, 5.0)
        val b = Vector(4.0, 7.0, 10.0)
        val degree = a.cosToRadian(a.cos(b))
        assertEquals(expectedDegree, degree)
    }

    @Test
    fun add() {
        val expectedValue = 16.0
        val a = Vector(19.0, 18.0, 17.0)
        val b = Vector(-3.0, -2.0, -1.0)
        val result = a.add(b)
        assertEquals(expectedValue, result.x)
        assertEquals(expectedValue, result.y)
        assertEquals(expectedValue, result.z)
    }

    @Test
    fun sub() {
        val expectedX = 0.0
        val expectedY = 5.0
        val expectedZ = 15.0
        val a = Vector(-1.0, 3.0, 6.0);
        val b = Vector(-1.0, -2.0, -9.0);
        val result = a.sub(b)
        assertEquals(expectedX, result.x)
        assertEquals(expectedY, result.y)
        assertEquals(expectedZ, result.z)
    }

    @Test
    fun generateRandomVectors() {
        val vectorsCount = 5
        val vector = Vector
        val generateRandomVectors = vector.generateRandomVectors(5)
        assertEquals(vectorsCount, generateRandomVectors.size)
    }
}